services/movie_session.py

```
@@ -0,0 +1,57 @@
+from django.db.models import QuerySet
+
+from db.models import MovieSession
+from datetime import datetime
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int,
+) -> MovieSession:
+    movie_session = MovieSession.objects.create(
+        show_time=movie_show_time,
+        cinema_hall_id=cinema_hall_id,
+        movie_id=movie_id,
+    )
+    return movie_session
+
+
+def get_movies_sessions(session_date: str | None = None) -> QuerySet:
+    if session_date:
+        movie_sessions = MovieSession.objects.filter(
+            show_time__date=session_date
+        )
+    else:
+        movie_sessions = MovieSession.objects.all()
+
+    return movie_sessions
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(pk=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: datetime | None = None,
+        movie_id: int | None = None,
+        cinema_hall_id: int | None = None,
+
+) -> MovieSession:
+    movie_session = MovieSession.objects.get(pk=session_id)
+
+    if show_time is not None:
+        movie_session.show_time = show_time
+    if movie_id is not None:
+        movie_session.movie_id = movie_id
+    if cinema_hall_id is not None:
+        movie_session.cinema_hall_id = cinema_hall_id
+
+    movie_session.save()
+
+    return movie_session
+
+
+def delete_movie_session_by_id(session_id: int) -> None:
+    MovieSession.objects.filter(id=session_id).delete()
```

This violates the principle of checklist item #Code Efficiency.3. When deleting a single object by its unique ID, it's better practice to use `.get()` to retrieve the object and then call `.delete()` on it. This makes your intent clearer and will raise a `DoesNotExist` error if the session ID is invalid, which is generally more robust than failing silently.